debvader/deblender.py

```diff
@@ -82,37 +83,61 @@ def get_residual_field(self, res_deblend=None):
-                output_images_mean_padded = np.zeros(
-                    (1, self.field_size, self.field_size, self.nb_of_bands)
-                )
-                pos_offset = int((self.field_size - self.cutout_size) / 2)
-
-                output_images_mean_padded[
-                    :,
-                    pos_offset : self.cutout_size + pos_offset,
-                    pos_offset : self.cutout_size + pos_offset,
-                    :,
-                ] = isolated_galaxy_row["output_images_mean"]
 
                 # subtract the image of the currently galaxy after positioning it correctly
                 # TODO: try to avoid using scipy.ndimage.shift here, it is super slow. Diectly subtract at correct index instead of creating a padding?
-                x_pos = (
-                    isolated_galaxy_row["galaxy_distances_to_center_x"]
-                    + isolated_galaxy_row["shifts"][0]
-                )
-                y_pos = (
-                    isolated_galaxy_row["galaxy_distances_to_center_y"]
-                    + isolated_galaxy_row["shifts"][1]
-                )
 
-                for band in range(self.nb_of_bands):
-                    deblended_image[0, :, :, band] -= scipy.ndimage.shift(
-                        output_images_mean_padded[0, :, :, band], shift=(x_pos, y_pos)
+                pos_offset = int((self.field_size - self.cutout_size) / 2)
+
+                if self.optimize_positions:
+
+                    # First create padded images of the stamps at the size of the field to allow for a simple subtraction.
+
+                    output_images_mean_padded = np.zeros(
+                        (1, self.field_size, self.field_size, self.nb_of_bands)
+                    )
+
+                    output_images_mean_padded[
+                        :,
+                        pos_offset : self.cutout_size + pos_offset,
+                        pos_offset : self.cutout_size + pos_offset,
+                        :,
+                    ] = isolated_galaxy_row["output_images_mean"]
+
+                    x_pos = (
+                        isolated_galaxy_row["galaxy_distances_to_center_x"]
+                        + isolated_galaxy_row["shifts"][0]
+                    )
+                    y_pos = (
+                        isolated_galaxy_row["galaxy_distances_to_center_y"]
+                        + isolated_galaxy_row["shifts"][1]
+                    )
+
+                    for band in range(self.nb_of_bands):
+                        deblended_image[0, :, :, band] -= scipy.ndimage.shift(
+                            output_images_mean_padded[0, :, :, band],
+                            shift=(x_pos, y_pos),
+                        )
+                else:
+                    x_start = int(
+                        round(
+                            isolated_galaxy_row["galaxy_distances_to_center_x"]
+                            + pos_offset
+                        )
```

Comment:
When using `round` it is common to specify the decimal until which you **choose** to round the value.